domain/src/main/java/org/oppia/android/domain/exploration/ExplorationDataController.kt

```diff
@@ -80,6 +85,38 @@ class ExplorationDataController @Inject constructor(
     }
   }
 
+  /**
+   * Checks the checkpoint state to make sure that checkpointing has been successful up-to that
+   * point in the exploration.
+   *
+   * @return a one-time [LiveData] with success result if checkpointing has been successful
+   *         otherwise a failure result with an appropriate exception is returned.
+   */
+  fun checkHasCheckpointingBeenSuccessful(): LiveData<AsyncResult<Any?>> {
+    return try {
+      explorationProgressController.isCurrentCheckpointStateIsSavedDatabaseNotExceededLimit()
+      MutableLiveData(AsyncResult.success(null))
+    } catch (e: Exception) {
+      exceptionsController.logNonFatalException(e)
+      MutableLiveData(AsyncResult.failed(e))
+    }
+  }
+
+  /** Function to fetch the details of the oldest saved exploration for a specified profileId.*/
```

Comment:
Add `@param` and `@return` annotations (here and below)

Reply:
done